cms/conf.py

```diff
@@ -178,74 +214,107 @@ def __init__(self):
             self.cache_dir = os.path.join("/", "var", "local", "cache", "cms")
             self.data_dir = os.path.join("/", "var", "local", "lib", "cms")
             self.run_dir = os.path.join("/", "var", "local", "run", "cms")
-            paths = [os.path.join("/", "usr", "local", "etc", "cms.conf"),
-                     os.path.join("/", "etc", "cms.conf")]
+            etc_paths = [os.path.join("/", "usr", "local", "etc"),
+                     os.path.join("/", "etc")]
         else:
             self.log_dir = "log"
             self.cache_dir = "cache"
             self.data_dir = "lib"
             self.run_dir = "run"
-            paths = [os.path.join(".", "config", "cms.conf")]
+            etc_paths = [os.path.join(".", "config")]
             if '__file__' in globals():
-                paths += [os.path.abspath(os.path.join(
+                etc_paths += [os.path.abspath(os.path.join(
                           os.path.dirname(__file__),
-                          '..', 'config', 'cms.conf'))]
-            paths += [os.path.join("/", "usr", "local", "etc", "cms.conf"),
-                      os.path.join("/", "etc", "cms.conf")]
+                          '..', 'config'))]
+            etc_paths += [os.path.join("/", "usr", "local", "etc"),
+                      os.path.join("/", "etc")]
+
+        legacy_paths = [os.path.join(p, "cms.conf") for p in etc_paths]
+        paths = [os.path.join(p, "cms.toml") for p in etc_paths]
```

Comment:
I was thinking that maybe it would make sense here to not differentiate between `legacy_paths` and normal `paths`, we could just have `paths` (with the legacy ones appended at the end) and simply try to read sequentially each one of them: as soon as we find a file that we manage to read we return the configuration found there. We would try to read in JSON basically only when reading in TOML raises some kind of format exception. I think the code would be simpler. What do you think?

Reply:
I agree. I put the legacy paths first though, in case the new sample config is copied during installation: Otherwise, this sample config would be used inadvertently and the message about the old config file would not appear. So now,
* a warning is going to be raised as soon as one of the files can be read in JSON format before one can be read in TOML format, and
* this JSON config will then proceed to be loaded.

Let me know if I should change this.